stats.py

```diff
@@ -0,0 +1,41 @@
+def count_words(text):
+    counter = 0
+    lines = text.split('\n') 
+    for line in lines:
+        words = line.split(' ')
+        for word in words:
+            if len(word) > 0:
+                counter += 1
+    return counter
+
+def sort_on(items):
+    return items["count"]
+
+def sort_chars(d):
+    sorted_chars = []
+    for key in d:
+        if key.isalpha():
+            new_d = {'char':key, 'count': d[key]}
+            sorted_chars.append(new_d)
+    sorted_chars.sort(reverse=True,key=sort_on)
+    sorted_dictionary = {}
+    for dictionary in sorted_chars:
+        key = dictionary['char']
+        value = dictionary['count']
+        sorted_dictionary[key] = value
+    return sorted_dictionary
+
+def count_chars(text): 
+    words = text.split()
+    counter = 0
+    dic = {}
+    for word in words:
+        chars = list(word)
+        for char in chars:
+            char = char.lower()
+            if char in dic:
+                dic[str(char)] += 1
+            else:
+                dic[str(char)] = 1
+                counter += 1
+    return dic
```

Comment:
No need to initialize `counter`. We can just pass it inline. When there's already a key for said `char` we can just increment by 1. When there's no key we add a key with initial value of 1.
```python
text = 'whole text of frankenstein.txt'
def count_chars(text):
	dictionary = {}
	for char in text:
		lowered_char = char.lower()
		if lowered_char in dictiohary:
			dictionary[lowered_char] += 1
		else:
			dictionary[lowered_char] = 1
			
	return dictionary
```